File: Minimum Waiting Time/O(nlogn)_time_and_O(1)_space.py
def minimumWaitingTime(queries):
    # Write your code here.
    queries.sort()
    minwaitingTime=0
    for ind, duration in enumerate(queries):
        queriesLeft = len(queries) - ind - 1
        minwaitingTime += duration*queriesLeft
    return minwaitingTime

def minimumWaitingTime1(queries):
    # Write your code here.
    queries.sort()
    minWaitingTime=0
    prevTimes=0
    for query in queries:
        minWaitingTime+=prevTimes
        prevTimes+=query
    return minWaitingTime

File: Greedy/Minimum Waiting Time/O(nlogn)_time_and_O(1)_space.js
function minimumWaitingTime(queries) {
  // Write your code here.
  queries.sort((a, b) => a - b);
  minWaitingTime = 0;
  for (let i = 0; i < queries.length; i++) {
    let currentValue = queries[i];
    let remainingQueries = queries.length - i - 1;
    minWaitingTime += currentValue * remainingQueries;
  }
  return minWaitingTime;
}

function minimumWaitingTime(queries) {
  // Write your code here.
  queries.sort((a, b) => a - b);
  let currentSum = 0;
  let previousSum = 0;
  for (let query of queries) {
    currentSum += previousSum;
    previousSum += query;
  }
  return currentSum;
}
